source/common/router/router.cc

```diff
@@ -1581,6 +1590,12 @@ bool Filter::convertRequestHeadersForInternalRedirect(Http::RequestHeaderMap& do
     }
   }
 
+  // See https://tools.ietf.org/html/rfc7231#section-6.4.4.
+  if (status_code == enumToInt(Http::Code::SeeOther)) {
```

Comment:
Arguably this gets runtime guarded but given bodies weren't previously supported, 303s aren't the norm, and this is moving to spec compliance, I'm inclined to say it's better to avoid the extra complexity.  Cc @snowp for second thumbs up.

Reply:
> given bodies weren't previously supported, 303s aren't the norm, and this is moving to spec compliance

yeah that was my thinking, and if the runtime guard is disabled + a body is present then we wouldn't call this function anyways due to some logic on line 1495 that short-circuits to not do the internal redirect, but I can add the runtime guard here if desired.

Reply:
Yeah I think if we're just supporting something that used to be unsupported previously a runtime flag won't be necessary. I doubt anyone would be relying on the lack of support for 303s.